app/containers/HomePage/reducer.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * HomeReducer
+ *
+ * The reducer takes care of our data. Using actions, we can change our
+ * application state.
+ * To add a new action, add it to the switch statement in the reducer function
+ *
+ * Example:
+ * case YOUR_ACTION_CONSTANT:
+ *   return state.set('yourStateVariable', true);
+ */
+
+import {
+  CHANGE_USERNAME,
+  LOAD_REPOS_SUCCESS,
+  LOAD_REPOS,
+  LOAD_REPOS_ERROR,
+} from './constants';
+import { fromJS } from 'immutable';
+
+// The initial state of the App
+const initialState = fromJS({
+  loading: false,
+  error: false,
+  currentUser: false,
+  userData: fromJS({
+    repositories: false,
+    username: '',
+  }),
+});
+
+function homeReducer(state = initialState, action) {
+  switch (action.type) {
+    case CHANGE_USERNAME:
+      // Delete prefixed '@' from the github username
+      return state
+        .setIn(['userData', 'username'], action.name.replace(/@/gi, ''));
+    case LOAD_REPOS:
```

Comment:
Can we move the global `LOAD_REPOS`, `LOAD_REPOS_SUCCESS` and `LOAD_REPOS_ERROR` actions and constants back to the `App` component, only leaving the `CHANGE_USERNAME` in this reducer?